dapp_runner/runner/runner.py

```diff
@@ -91,27 +137,7 @@ async def start(self):
         await self._create_networks()
         await self._load_payloads()
 
-        for service_name, service_descriptor in self.dapp.nodes.items():
-            cluster_class, run_params = await get_service(
-                service_name, service_descriptor, self._payloads, self._networks
-            )
-            cluster = await self.start_cluster(service_name, cluster_class, run_params)
-            if service_descriptor.http_proxy:
-                await self._start_local_proxy(
-                    service_name, cluster, service_descriptor.http_proxy.ports[0]
-                )
-
-            # launch queue listeners for all the service instances
-            for idx in range(len(cluster.instances)):
-                s = cluster.instances[idx]
-                self._tasks.extend(
-                    [
-                        asyncio.create_task(self._listen_state_queue(s)),
-                        asyncio.create_task(
-                            self._listen_data_queue(service_name, idx, s)
-                        ),
-                    ]
-                )
+        self._tasks.append(asyncio.create_task(self._start_services()))
```

Comment:
I don't understand why do we await for all the other things in this method, but create a task for this particular one. I'm not saying this is wrong, just only that I don't understand.

Also: the docstring could include this - something like "Start the Golem engine, create networks, create payloads and initialize the task that starts the dapp".

Reply:
yup, your description is accurate and here's additional explanation (I can include as a comment if you think that's useful):

"start golem", `create_networks` and `create_payloads` are:
1. almost instantaneous
2. are - right now at least - indispensible before we can start starting the services... 

thus, I believe it makes sense to await until they're finished since we cannot do - or report - anything useful to the outside world until those are finished... 

earlier, when the creation of clusters happened without any dependencies between them, we could also just immediately start all the clusters and only then start tracking the app state... 

now, we don't want to await that since then we'll be losing an important part of the state transitions (which an external user of `dapp_runner` may wish to track) from starting to spawn instances to finalization of the whole initialization of the dapp, when some of the services need to wait for their turn before others have completed their initialization...

Reply:
OK, cool, this makes sense :)

Please add a comment (single line like "we start services in a separate task, so that the service-starting progress can be tracked" would be enough)

Reply:
:+1: